hw1_2/Program.cs

```diff
@@ -0,0 +1,197 @@
+                ref indexOfLastLetterInOriginalString);
+            string transformedString = "";
+            foreach (var tuple in arrayOfSuffixes)
+                transformedString += tuple.Item2; //concatenation of sorted suffixes prior 
+            endIndexOfOriginalInTransformedString = indexOfLastLetterInOriginalString;
+            transformedString =
+                transformedString.Replace((char)1, lastSymbol); //change our convenient symbol to original
+            return transformedString;
+        }
+
+        static (char, int)[] ArrayOfDifferentLettersInSortedStringWithTheirNumberInItConstructor(string sortedString,
+            out int currIndex)
+        {
+            (char, int)[] uniqueLettersAndTheirAmount = new (char, int) [sortedString.Length];
+            uniqueLettersAndTheirAmount[0] = (sortedString[0], 1);
+            currIndex = 0;
+            for (int i = 1; i < sortedString.Length; i++)
+            {
+                if (sortedString[i] != sortedString[i - 1])
+                {
+                    currIndex++;
+                    uniqueLettersAndTheirAmount[currIndex].Item1 = sortedString[i];
+                }
+
+                uniqueLettersAndTheirAmount[currIndex].Item2++;
+            }
+
+            return uniqueLettersAndTheirAmount;
+        }
+
+        static string BurrowsWheelerRetransform(string transformedString, int endPositionOfOriginalInTransformedString)
+        {
+            //change last symbol with symbol that will indicate the end of the string because it solves problem of sort in sortedDifferentLettersInTransformedString
+            char lastSymbol =
+                transformedString[
+                    endPositionOfOriginalInTransformedString -
+                    1]; //where symbol of the end have to be in first position
+            transformedString = transformedString.Remove(endPositionOfOriginalInTransformedString - 1, 1)
+                .Insert(endPositionOfOriginalInTransformedString - 1, ((char)1).ToString());
+
+            int transformedStringLenght = transformedString.Length;
+            string sortedTransformedString = "";
+            for (int i = 0; i < transformedStringLenght; i++)
+            {
+                sortedTransformedString += transformedString[i];
+            }
+
+            sortedTransformedString = string.Concat(sortedTransformedString.OrderBy(x => x).ToArray());
+            int lastIndexWithLetter = 0;
+            (char, int)[] uniqueLettersAndTheirAmount =
+                ArrayOfDifferentLettersInSortedStringWithTheirNumberInItConstructor(sortedTransformedString,
+                    out lastIndexWithLetter);
+            string sortedDifferentLettersInTransformedString = "";
+
+            for (int i = 0; i <= lastIndexWithLetter; i++)
+            {
+                sortedDifferentLettersInTransformedString += uniqueLettersAndTheirAmount[i].Item1;
+            }
+
+            int[] a = new int[lastIndexWithLetter + 1];

```

Comment:
```suggestion
            var a = new int[lastIndexWithLetter + 1];
```
Однобуквенные имена переменных - так нельзя.